src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,552 @@
+}
+
+type broker struct {
+	Id   int32  `json:"id"`
+	Addr string `json:"addr"`
+}
+
+type topic struct {
+	Name          string             `json:"name"`
+	Config        map[string]*string `json:"config"`
+	NumPartitions int32              `json:"numPartitions"`
+	ReplFactor    int16              `json:"replFactor"`
+}
+
+type consumerGroup struct {
+	Id      string                 `json:"id"`
+	Members []*consumerGroupMember `json:"members"`
+}
+
+type consumerGroupMember struct {
+	Host string `json:"host"`
+	Id   string `json:"id"`
+}
+
+func saveKafkaMetadata(ps *stepParams, admin sarama.ClusterAdmin) step {
+	return func() error {
+		log.Debug("Reading Kafka metadata")
+		bs, _, err := admin.DescribeCluster()
+		if err != nil {
+			return fmt.Errorf("couldn't list topics: %w", err)
+		}
+
+		brokers := make([]*broker, 0, len(bs))
+		for _, b := range bs {
+			brokers = append(brokers, &broker{
+				Id:   b.ID(),
+				Addr: b.Addr(),
+			})
+		}
+
+		topicsMap, err := admin.ListTopics()
+		if err != nil {
+			return fmt.Errorf("couldn't list topics: %w", err)
+		}
+		topics := make([]*topic, 0, len(topicsMap))
+		for topicName, details := range topicsMap {
+			n, ds := topicName, details
+			topics = append(topics, &topic{
+				Name:          n,
+				Config:        ds.ConfigEntries,
+				NumPartitions: ds.NumPartitions,
+				ReplFactor:    ds.ReplicationFactor,
+			})
+		}
+		cgs2proto, err := admin.ListConsumerGroups()
+		if err != nil {
+			return fmt.Errorf("couldn't list consumer groups: %w", err)
+		}
+		cgNames := make([]string, 0, len(cgs2proto))
+		for name := range cgs2proto {
+			cgNames = append(cgNames, name)
+		}
+		cgDescs, err := admin.DescribeConsumerGroups(cgNames)
+		if err != nil {
+			if len(cgDescs) == 0 {
+				return fmt.Errorf("couldn't list describe consumer groups: %w", err)
+			}
+		}
+		cgs := make([]*consumerGroup, 0, len(cgDescs))
+		for _, desc := range cgDescs {
+			d := desc
+			members := make([]*consumerGroupMember, 0, len(d.Members))
+			for _, m := range d.Members {
+				members = append(members, &consumerGroupMember{
+					Host: m.ClientHost,
+					Id:   m.ClientId,
+				})
+			}
+			cgs = append(cgs, &consumerGroup{
+				Id:      d.GroupId,
+				Members: members,
+			})
+		}
+		bytes, err := json.Marshal(&kafkaMetadata{
+			Brokers:        brokers,
+			Topics:         topics,
+			ConsumerGroups: cgs,
+		})
+		if err != nil {
+			return fmt.Errorf("couldn't marshal Kafka metadata: %w", err)
+		}
+		return writeFileToZip(ps, "kafka.json", bytes)
+	}
+}
+
+// TODO: filter by time or size (GiB)
+// TODO: What if running inside a container/ k8s?
+func saveLogs(ps *stepParams, since, until string) step {
+	return func() error {
+		args := []string{"--no-pager", "-u", "redpanda"}
+		if since != "" {
+			args = append(args, "--since", since)
+		}
+		if until != "" {
+			args = append(args, "--until", until)
+		}
+		return writeCommandOutputToZip(
+			ps,
+			"redpanda.log",
+			"journalctl",
```

Comment:
Do we have concerns about bundling sensitive information here? I don't think redpanda dumps sensitive info, just checking.